File: src/main/java/com/agentstore/common/exception/client/DomainClientException.java
package com.agentstore.common.exception.client;

import com.agentstore.common.exception.constants.ErrorCode;

public class DomainClientException extends ClientException {
    public DomainClientException(ErrorCode errorCode, Object... messageArguments) {
        super(errorCode, messageArguments);
    }

    public DomainClientException(ErrorCode errorCode, Throwable cause, Object... messageArguments) {
        super(errorCode, cause, messageArguments);
    }
}

File: src/main/kotlin/com/agentstore/common/exception/client/DomainClientException.kt
package com.agentstore.common.exception.client

import com.agentstore.common.exception.constants.ErrorCode

class DomainClientException : ClientException {
    constructor(errorCode: ErrorCode, vararg messageArguments: Any?) : super(
        errorCode,
        *messageArguments,
    )

    constructor(
        errorCode: ErrorCode,
        cause: Throwable,
        vararg messageArguments: Any?,
    ) : super(errorCode, cause, *messageArguments)
}
